circuit/src/contracts-registry/src/benchmarking.rs

```diff
@@ -0,0 +1,250 @@
+        let requester: T::AccountId = account("TEST", 1_u32, USER_SEED);
+    }: _(RawOrigin::Root, requester.clone(), test_contract.clone())
+    verify {
+        assert!(pallet::ContractsRegistry::<T>::contains_key(&contract_id));
+    }
+
+    purge {
+        let requester_1: T::AccountId = account("TEST", 1_u32, USER_SEED);
+
+        let test_contract_1: RegistryContract<T::Hash, T::AccountId, BalanceOf<T>, T::BlockNumber> = RegistryContract {
+            code_txt: b"some_code".to_vec(),
+            bytes: vec![],
+            author: account("TEST", 1_u32, USER_SEED),
+            author_fees_per_single_use: None,
+            abi: None,
+            action_descriptions: vec![],
+            info: None,
+            meta: ContractMetadata::new(
+                vec![],
+                b"contract 1".to_vec(),
+                vec![],
+                vec![],
+                None,
+                None,
+                None,
+                None,
+                None,
+            ),
+        };
+
+        let requester_2: T::AccountId = account("TEST", 2_u32, USER_SEED);
+
+        let test_contract_2: RegistryContract<T::Hash, T::AccountId, BalanceOf<T>, T::BlockNumber> = RegistryContract {
+            code_txt: b"some_code_2".to_vec(),
+            bytes: vec![],
+            author: account("TEST", 2_u32, USER_SEED),
+            author_fees_per_single_use: None,
+            abi: None,
+            action_descriptions: vec![],
+            info: None,
+            meta: ContractMetadata::new(
+                vec![],
+                b"contract 2".to_vec(),
+                vec![],
+                vec![],
+                None,
+                None,
+                None,
+                None,
+                None,
+            ),
+        };
+
+        let contract_id_1 = test_contract_1.generate_id::<T>();
+        crate::ContractsRegistry::<T>::insert(
+            test_contract_1.generate_id::<T>(),
+            test_contract_1.clone(),
+        );
+
+        let contract_id_2 = test_contract_2.generate_id::<T>();
+        crate::ContractsRegistry::<T>::insert(
+            test_contract_2.generate_id::<T>(),
+            test_contract_2.clone(),
+        );
+    }: _(RawOrigin::Root, requester_1.clone(), contract_id_1.clone())
+    verify {
+        assert_ok!(ContractsRegistry::<T>::purge(RawOrigin::Root.into(), requester_2, contract_id_2));
+    }
+
+    fetch_contracts {
+        let test_contract_author_1: RegistryContract<T::Hash, T::AccountId, BalanceOf<T>, T::BlockNumber> = RegistryContract {
+            code_txt: b"some_code_1".to_vec(),
+            bytes: vec![],
+            author: account("TEST", 1_u32, USER_SEED),
+            author_fees_per_single_use: None,
+            abi: None,
+            action_descriptions: vec![],
+            info: None,
+            meta: ContractMetadata::new(vec![], vec![], vec![], vec![], Some(b"contract 1".to_vec()), None, None, None, None),
+        };
```

Comment:
I see that same object is being used again and again with minimal changes.

Can we attempt to create a function which returns a dummy `RegistryContract` object when called.
This will clean up the code significantly.